deploy/lib/deployer/repo.rb

```diff
@@ -1,29 +1,40 @@
+require 'byebug'
 class Deployer
   class Repo
-    def initialize(name, package_name:, config:, updater: nil, config_file: ConfigFile.new(name, config: config))
+    def initialize(name, package_name:, config:, updater: nil, config_file: ConfigFile.new(name, config: config), dependabot_proxy: Deployer::Repo::DependabotProxy.new(name, config: config, package_name: package_name))
       @name = name
       @config = config
       @package_name = package_name
       @updater = updater || default_updater
       @config_file = config_file
+      @status = :pending
+      @dependabot_proxy = dependabot_proxy
     end
 
     def update_package
+      return skipped! unless attempt_to_update?
       updater.run
 
+      if updater.skipped
+        self.status = :skipped
+        @repo_upgrade_status = :version_bump_not_possible
+        return
```

Comment:
Is this the Dependabot update process itself? It goes to update and sees it's a major version bump, and the repo has a rule to only do MINOR or PATCH, so it chooses to skip?